Purchases/Classes/Public/RCPurchaserInfo.m

```diff
@@ -92,16 +92,28 @@ - (instancetype _Nullable)initWithData:(NSDictionary *)data
 {
     NSMutableSet *activeSubscriptions = [NSMutableSet setWithCapacity:dates.count];
 
-    for (NSString *productIdentifier in dates) {
-        NSObject *dateOrNull = dates[productIdentifier];
-        if ([dateOrNull isKindOfClass:NSNull.class] || ((NSDate *)dateOrNull).timeIntervalSinceNow > 0) {
-            [activeSubscriptions addObject:productIdentifier];
+    for (NSString *identifier in dates) {
+        NSDate *dateOrNull = (NSDate *)dates[identifier];
+        if ([dateOrNull isKindOfClass:NSNull.class] || [self isActive:dateOrNull]) {
+            // If the expiration date is not null and the day we fetched this is before the expiration,
+            // then this is an active subscription
```

Comment:
I think you can kill this comment. 

If the code is written well and the names are clear, these comments don't really add much. I think that is the case here.